layouts/partials/functions/link.html

```diff
@@ -0,0 +1,48 @@
+<!--
+Prepare a properly internationalized link. Returns a properly resolved (translated) URL for a given slug. Will fall back
+to the English version (and suffix the link text accordingly) if a translated version doesn't exist.
+
+Use it like this (remove the `.`):
+
+    ```
+    {.{ $link := (partial "functions/link" (dict "slug" "/generator" "text" "Request generator" }}
+    <a href="{.{ $link.url }}">{.{ $link.text }}</a>
+    ```
+
+With the following parameters:
+    - `[slug]`: The (untranslated) slug of the page to be referenced.
+    - `[text]`: The desired link text. Either as the actual text (if `translate == false`) or as a translation key (if
+                `translate == true`).
+    - `[fragment]`: An (optional) document fragment to append to the URL. Don't add the `#`.
+    - `[translate]`: Whether to interpret `text` as a translation key and replace it with the respective translated text
+                     (`true`) or as literal text (`false`). Defaults to `false`.
+
+Returns a dict with the following properties:
+
+    - `[url]`: The resolved absolute URL.
+    - `[text]`: The resulting link text.
+-->
+
+{{ $translate := .translate }}
+{{ if not .translate }}{{ $translate = false }}{{ end }}
+
+{{ $page := (site.GetPage .slug) }}
+{{ $text := (cond $translate (T .text .) .text) }}
+{{ if not $page }}
+    <!-- If the page doesn't exist in our current language, we fall back to English and apply a suffix to the text. -->
+    <!-- This depends on English being at index 1, which is highly unlikely to ever change. The alternative would have
+         been an ugly loop. -->
+    {{ $page = ((index site.Sites 1).GetPage .slug) }}
+    {{ $text = (printf "%s (EN)" $text) }}
```

Comment:
I am thinking this suffix should be wrapped in a `<span>` so that we can style it differently.

Reply:
Hm, the problem is that we can't do that clean here. We could only put raw HTML into the string and would then have to remember to `markdownify` that downstream for all users.

Do you have a specific styling in mind? Otherwise I would propose to postpone that change until someone actually wants to style it.

Reply:
I am fine with postponing, I just wanted to note this.